### PR TITLE
refactor(iec/keypair): adjust code style

### PR DESCRIPTION
**Pull request by ruwenqiang123**
**What this PR does / why we need it**:
move the file(resource_huaweicloud_iec_keypair) and fix error and  lint this resource
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fix some non-conformance functions and parameter names
**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
root@ecs-ruwenqiang:/home/huawei/go/src/github.com/huaweicloud/terraform-provider-huaweicloud# make testacc TEST="./huaweicloud/services/acceptance/iec" TESTARGS="-run TestAccIECKeypairResource_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iec -v -run TestAccIECKeypairResource_basic -timeout 360m -parallel 4
=== RUN   TestAccIECKeypairResource_basic
--- PASS: TestAccIECKeypairResource_basic (19.18s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iec       19.214s
```

### CODE REVIEW

File: huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go [hunk 1]
```diff
@@ -45,58 +49,63 @@ func resourceIecKeypair() *schema.Resource {
...
-		return fmtp.Errorf("Error creating HuaweiCloud IEC client: %s", err)
+		return diag.Errorf("error creating IEC v1 client: %s", err)
 	}
 
 	err = keypairs.Delete(iecClient, d.Id()).ExtractErr()
 	if err != nil {
-		return fmtp.Errorf("Error deleting HuaweiCloud iec keypair: %s", err)
+		return diag.Errorf("error deleting IEC keypair: %s", err)
 	}
 	d.SetId("")
```
Comment (Lance52259) [inline huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go]: `d.SetId("")`
there is unnecessary to support this function in delete method.

Comment (ruwenqiang123) [reply huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go]: 已修改

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_keypair_test.go [hunk 2]
```diff
@@ -42,63 +60,10 @@ func TestAccIECKeypairResource_basic(t *testing.T) {
...
-		}
-
-		*kp = *found
-
-		return nil
-	}
-}
-
-func testAccIECKeypair_basic(rName string) string {
+func testAccIECKeypair_basic(name string) string {
```
Comment (Lance52259) [inline huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_keypair_test.go]:
```suggestion
func testAccKeypair_basic(name string) string {
```

File: huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go [hunk 3]
```diff
@@ -45,59 +49,63 @@ func resourceIecKeypair() *schema.Resource {
...
+	iecClient, err := cfg.IECV1Client(cfg.GetRegion(d))
 	if err != nil {
-		return fmtp.Errorf("Error creating HuaweiCloud IEC client: %s", err)
+		return diag.Errorf("error creating IEC v1 client: %s", err)
 	}
 
 	kp, err := keypairs.Get(iecClient, d.Id()).Extract()
 	if err != nil {
-		return CheckDeleted(d, err, "keypair")
+		return common.CheckDeletedDiag(d, err, "IEC keypair, missing service classification")
```
Comment (Lance52259) [inline huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go]:
```suggestion
		return common.CheckDeletedDiag(d, err, "IEC keypair")
```

File: huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_keypair_test.go (hunk 2 above)
Comment (ruwenqiang123) [reply huaweicloud/services/acceptance/iec/resource_huaweicloud_iec_keypair_test.go]: have changed

File: huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go (hunk 3 above)
Comment (ruwenqiang123) [reply huaweicloud/services/iec/resource_huaweicloud_iec_keypair.go]: have changed

**Pull request merged**